Objective-C/CBLConsoleLogger.m

```diff
@@ -43,9 +46,51 @@ - (void) logWithLevel: (CBLLogLevel)level domain: (CBLLogDomain)domain message:
     if (self.level > level || (self.domains & domain) == 0)
         return;
     
-    NSString* levelName = CBLLog_GetLevelName(level);
-    NSString* domainName = CBLLog_GetDomainName(domain);
-    NSLog(@"CouchbaseLite %@ %@: %@", domainName, levelName, message);
+    os_log_t osLogDomain = osLogDictionary[@(domain)];
+    os_log_type_t osLogType = osLogTypeForLevel(level);
+    os_log_with_type(osLogDomain, osLogType, "%@", message);
+}
+
+static os_log_type_t osLogTypeForLevel(CBLLogLevel level) {
+    switch (level) {
+        case kCBLLogLevelDebug:
+            return OS_LOG_TYPE_DEBUG;
+        case kCBLLogLevelVerbose:
+            return OS_LOG_TYPE_INFO; // Map verbose to info
+        case kCBLLogLevelInfo:
+            return OS_LOG_TYPE_INFO;
+        case kCBLLogLevelWarning:
+            return OS_LOG_TYPE_ERROR; // Map warning to error
+        case kCBLLogLevelError:
+            return OS_LOG_TYPE_ERROR;
+        default:
+            return OS_LOG_TYPE_DEFAULT; // Default log type
+    }
+}
+
+- (void) initializeOSLogDomains {
+    osLogDictionary = [NSMutableDictionary dictionary];
+        
+    osLogDictionary[@(kCBLLogDomainDatabase)] = os_log_create("com.couchbase.lite.ios", "Database");
```

Comment:
We should also get "com.ouchbase.lite.ios" from the framework bundle identifier instead of hard code the string here.